loadtest/loadtest.go

```diff
@@ -0,0 +1,328 @@
+package loadtest
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/apprclient"
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/e2e-harness/pkg/framework"
+	"github.com/giantswarm/helmclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/afero"
+	yaml "gopkg.in/yaml.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/helm/pkg/helm"
+)
+
+type Config struct {
+	Clients        *Clients
+	GuestFramework *framework.Guest
```

Comment:
`GuestFramework` is only used for tenant cluster clients. Let's not use `e2e-harness` here for that as we want to phase it out. I think we should have something like this. 

```
CPClients *k8s.Clients
TCClients *k8s.Clients
```

Here `k8s` is https://github.com/giantswarm/e2esetup/blob/master/k8s. When you need the helm client we should add it to the clients there. We also have https://github.com/giantswarm/tenantcluster and should somehow consolidate the two packages.

Reply:
The problem is I need a TC client to install the test app in the TC.

I tried passing the TC client in as part of the config but it fails because the TC had not been launched yet.

Reply:
Can you elaborate? How injecting `framework.Guest` makes the TC being already launched?

Reply:
I think the problem Ross encountered was that the legacy guest framework waits for the api secret to be available. This secret holds the certs we use to access the tenant cluster. When the integration test starts and the e2e test here is executed, the api secret is not yet there so the initialization magic in the legacy guest framework does not yet have the clients prepared. We need to fix this somehow. At best the e2e tests might only be executed when all configuration is available and get be used. We need to think how this should look like.

Reply:
It's because in `framework.Guest` it waits for `guest.Setup` to complete. Without that the test will fail because the client has not been populated yet.

https://github.com/giantswarm/e2e-harness/blob/94388dde639308b70027f93cee0ae9012c989a6c/pkg/framework/guest.go#L78-L83

Reply:
WaitForAPIUp does this: https://github.com/giantswarm/e2e-harness/blob/94388dde639308b70027f93cee0ae9012c989a6c/pkg/framework/guest.go#L194

I guess we can repeat the functionality here or add it to e2esetup in general.

Reply:
@kopiczko Thanks. I'll take another look at this when I get time to work on it again.